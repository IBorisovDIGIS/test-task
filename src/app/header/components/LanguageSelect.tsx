import React, { useState } from 'react';
import { MenuItem, Select } from '@material-ui/core';
import { useTranslation } from 'react-i18next';
import { languages } from '../../../constants';

const LanguageSelect: React.FC = () => {
  const { i18n } = useTranslation();
  const [language, setLanguage] = useState(languages[0]);

  const handleSetLanguage = async (event: React.ChangeEvent<{ value: unknown }>) => {
    const lang = event.target.value as string;
    await i18n.changeLanguage(lang);
    setLanguage(lang);
  };

  return (
    <Select onChange={handleSetLanguage} value={language}>
      {languages.map((lang) => <MenuItem value={lang} key={lang}>{lang.toUpperCase()}</MenuItem>)}
    </Select>
  );
};

export default LanguageSelect;
