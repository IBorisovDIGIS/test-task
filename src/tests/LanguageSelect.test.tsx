import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {
  Select,
} from '@material-ui/core';
import toJson from 'enzyme-to-json';
import LanguageSelect from "../app/header/components/LanguageSelect";

configure({ adapter: new Adapter() });

describe('Language select should render correctly', () => {
  it('Should renders correctly', () => {
    const container = shallow(<LanguageSelect />);

    expect(toJson(container)).toMatchSnapshot();
  });

  it('Should render elements correctly', () => {
    expect(Select).toBeDefined();
  });

  it('Should change language', () => {
    const container = shallow(<LanguageSelect/>);

    container.find(Select).simulate('onChange', {
      target: { value: "ru" }
    });

    setTimeout(() => expect(container.find(Select).prop("value")).toBe('ru'))
  });

});
