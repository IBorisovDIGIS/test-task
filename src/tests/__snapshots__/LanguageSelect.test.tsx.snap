// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Language select should render correctly Should renders correctly 1`] = `
<WithStyles(ForwardRef(Select))
  onChange={[Function]}
  value="en"
>
  <WithStyles(ForwardRef(MenuItem))
    key="en"
    value="en"
  >
    EN
  </WithStyles(ForwardRef(MenuItem))>
  <WithStyles(ForwardRef(MenuItem))
    key="ru"
    value="ru"
  >
    RU
  </WithStyles(ForwardRef(MenuItem))>
  <WithStyles(ForwardRef(MenuItem))
    key="fr"
    value="fr"
  >
    FR
  </WithStyles(ForwardRef(MenuItem))>
</WithStyles(ForwardRef(Select))>
`;
